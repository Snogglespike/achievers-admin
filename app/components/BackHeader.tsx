import { Link } from "@remix-run/react";

import { ArrowLeft } from "iconoir-react";

interface Props {
  to?: string;
}

export function BackHeader({ to = "../" }: Props) {
  return (
    <>
      <div>
        <Link to={to} relative="path" className="btn-ghost btn mb-2 gap-2">
          <ArrowLeft className="w-6" />
          Back
        </Link>
      </div>

      <hr className="mb-4" />
    </>
  );
}
