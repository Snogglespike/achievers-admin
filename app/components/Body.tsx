import type { AzureUserWebAppWithRole } from "~/services";

import { Outlet } from "@remix-run/react";

import { Drawer } from "./Drawer";
import { Navbar } from "./Navbar";

interface Props {
  isAdmin: boolean;
  hasCompletedVolunteerAgreement?: boolean;
  version: string;
  currentUser: AzureUserWebAppWithRole;
}

export function Body({
  currentUser,
  hasCompletedVolunteerAgreement,
  isAdmin,
  version,
}: Props) {
  const showDrawer = isAdmin || hasCompletedVolunteerAgreement;

  return (
    <div className="drawer-mobile drawer">
      <input id="drawer" type="checkbox" className="drawer-toggle" />
      <div className="drawer-content flex flex-col">
        <Navbar currentUser={currentUser} />

        <main className="mt-16 flex h-full flex-col overflow-y-auto bg-white p-4">
          <Outlet />
        </main>
      </div>

      {showDrawer && <Drawer isAdmin={isAdmin} version={version} />}
    </div>
  );
}
