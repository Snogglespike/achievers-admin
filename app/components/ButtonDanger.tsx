import type { ButtonHTMLAttributes, DetailedHTMLProps } from "react";

export default function ButtonDanger({
  children,
  className = "",
  disabled,
  ...rest
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button
      {...rest}
      disabled={disabled}
      className={
        disabled
          ? "flex items-center justify-center rounded border bg-red-200 py-2 px-4 text-white " +
            className
          : "flex items-center justify-center rounded border bg-red-500 py-2 px-4 text-white hover:bg-red-600 " +
            className
      }
    >
      {children}
    </button>
  );
}
