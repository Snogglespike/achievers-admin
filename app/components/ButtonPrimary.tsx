import type { ButtonHTMLAttributes, DetailedHTMLProps } from "react";

export default function ButtonPrimary({
  children,
  className = "",
  disabled,
  ...rest
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button
      {...rest}
      disabled={disabled}
      className={
        disabled
          ? "flex items-center justify-center rounded bg-blue-200 py-2 px-4 text-white " +
            className
          : "flex items-center justify-center rounded bg-blue-500 py-2 px-4 text-white hover:bg-blue-600 " +
            className
      }
    >
      {children}
    </button>
  );
}
