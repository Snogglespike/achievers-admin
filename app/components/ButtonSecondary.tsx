import type { ButtonHTMLAttributes, DetailedHTMLProps } from "react";

export default function ButtonSecondary({
  children,
  className = "",
  disabled,
  ...rest
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button
      {...rest}
      disabled={disabled}
      className={
        disabled
          ? "flex items-center justify-center rounded bg-slate-100 py-2 px-4 text-slate-400 " +
            className
          : "flex items-center justify-center rounded border border-slate-300 bg-slate-200 px-4 py-2 hover:bg-slate-300 " +
            className
      }
    >
      {children}
    </button>
  );
}
