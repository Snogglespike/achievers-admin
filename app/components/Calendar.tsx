import type { Dayjs } from "dayjs";
import dayjs from "dayjs";

import { ArrowRightIcon, ArrowLeftIcon } from "@heroicons/react/24/solid";

interface Props {
  selectedDate: Dayjs;
  previousMonth: () => void;
  nextMonth: () => void;
  generateWeeksOfTheMonth: Date[][];
  currentDay: Date;
}

export default function Calendar({
  selectedDate,
  previousMonth,
  nextMonth,
  generateWeeksOfTheMonth,
  currentDay,
}: Props) {
  return (
    <div className="hidden w-full max-w-sm md:block">
      <div className="rounded-t bg-white p-5 md:p-8">
        <div className="flex items-center justify-between px-4">
          <span
            tabIndex={0}
            className="text-base font-bold text-gray-800 focus:outline-none"
          >
            {selectedDate.clone().format("YYYY MMM")}
          </span>
          <div className="flex items-center">
            <button
              aria-label="calendar backward"
              className="text-gray-800 hover:text-gray-400 focus:text-gray-400"
              onClick={previousMonth}
            >
              <ArrowLeftIcon className="w-6 text-blue-600" />
            </button>
            <button
              aria-label="calendar forward"
              className="ml-3 text-gray-800 hover:text-gray-400 focus:text-gray-400"
              onClick={nextMonth}
            >
              <ArrowRightIcon className="w-6 text-blue-600" />
            </button>
          </div>
        </div>
        <div className="flex items-center justify-between overflow-x-auto pt-12">
          <table className="w-full">
            <thead>
              <tr>
                {generateWeeksOfTheMonth[0].map((day, index) => (
                  <th key={`week-day-${index}`}>
                    <div className="flex w-full justify-center">
                      <p className="text-center text-base font-medium text-gray-800">
                        {dayjs(day).format("dd")}
                      </p>
                    </div>
                  </th>
                ))}
              </tr>
            </thead>
            <tbody>
              {generateWeeksOfTheMonth.map((week, weekIndex) => (
                <tr key={`week-${weekIndex}`}>
                  {week.map((day, dayIndex) => (
                    <td className="pt-6" key={`day-${dayIndex}`}>
                      {dayjs(currentDay).isSame(day, "date") ? (
                        <div className="h-full w-full">
                          <div className="flex w-full cursor-pointer items-center justify-center rounded-full">
                            <span
                              role="link"
                              tabIndex={0}
                              className="flex  h-8 w-8 items-center justify-center rounded-full bg-indigo-700 text-base font-medium text-white hover:bg-indigo-500 focus:bg-indigo-500 focus:outline-none focus:ring-2 focus:ring-indigo-700 focus:ring-offset-2"
                            >
                              {day.getDate()}
                            </span>
                          </div>
                        </div>
                      ) : (
                        <div className="flex w-full cursor-pointer justify-center px-2 py-2">
                          <p
                            className={
                              "text-base font-medium " +
                              (selectedDate.clone().toDate().getMonth() !==
                              day.getMonth()
                                ? "text-gray-400"
                                : "text-gray-700")
                            }
                          >
                            {day.getDate()}
                          </p>
                        </div>
                      )}
                    </td>
                  ))}
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}
