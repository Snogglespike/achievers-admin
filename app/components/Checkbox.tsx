interface Props {
  label?: string;
  name: string;
  defaultChecked?: boolean;
  required?: boolean;
  disabled?: boolean;
  readOnly?: boolean;
}

export function Checkbox({ label, name, required, ...props }: Props) {
  return (
    <div className="form-control relative">
      <label
        htmlFor={name}
        className="label cursor-pointer flex-col items-start"
      >
        <span className="label-text mb-1">{label}</span>
        {required && (
          <span className="label-text-alt absolute top-6 left-8 text-2xl text-error">
            *
          </span>
        )}
        <input
          type="checkbox"
          className="checkbox"
          id={name}
          name={name}
          required={required}
          {...props}
        />
      </label>
    </div>
  );
}
