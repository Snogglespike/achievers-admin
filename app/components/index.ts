export * from "./Checkbox";
export * from "./DateInput";
export * from "./Input";
export * from "./Select";

export * from "./Title";
export * from "./SubTitle";
export * from "./Text";

export * from "./LoadingSpinner";
export * from "./Navbar";
export * from "./Drawer";
