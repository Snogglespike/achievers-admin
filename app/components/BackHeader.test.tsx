import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";

import { BackHeader } from "~/components";

describe("BackHeader", () => {
  test("BackHeader snapshot", async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <BackHeader />
      </MemoryRouter>
    );

    expect(baseElement).toMatchSnapshot();
  });

  test("should have href defined via prop", async () => {
    render(
      <MemoryRouter>
        <BackHeader to="/test" />
      </MemoryRouter>
    );

    expect(screen.getByText("Back")).toHaveAttribute("href", "/test");
  });
});
