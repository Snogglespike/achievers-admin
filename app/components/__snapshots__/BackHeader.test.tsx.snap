// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackHeader > BackHeader snapshot 1`] = `
<body>
  <div>
    <div>
      <a
        class="btn-ghost btn mb-2 gap-2"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="w-6"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M20.25 12a.75.75 0 01-.75.75H6.31l5.47 5.47a.75.75 0 11-1.06 1.06l-6.75-6.75a.75.75 0 010-1.06l6.75-6.75a.75.75 0 111.06 1.06l-5.47 5.47H19.5a.75.75 0 01.75.75z"
            fill-rule="evenodd"
          />
        </svg>
        Back
      </a>
    </div>
    <hr
      class="mb-4"
    />
  </div>
</body>
`;
